spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/ReleaseAnalysisReport.java

```diff
@@ -51,25 +52,39 @@ public ReleaseAnalysisReport(List<String> applicationNamesToUpgrade, ReleaseDiff
 			Release existingRelease, Release replacingRelease) {
 		Assert.notNull(applicationNamesToUpgrade, "ApplicationNamesToUpgrade can not be null.");
 		Assert.notNull(releaseDifference, "ReleaseDifference can not be null.");
+		Assert.notNull(existingRelease, "ExistingRelease can not be null.");
+		Assert.notNull(replacingRelease, "ReplacingRelease can not be null.");
 		this.applicationNamesToUpgrade = applicationNamesToUpgrade;
 		this.releaseDifference = releaseDifference;
 		this.existingRelease = existingRelease;
 		this.replacingRelease = replacingRelease;
 	}
 
 	public List<String> getApplicationNamesToUpgrade() {
-		return applicationNamesToUpgrade;
+		return this.applicationNamesToUpgrade;
 	}
 
 	public ReleaseDifference getReleaseDifference() {
-		return releaseDifference;
+		return this.releaseDifference;
+	}
+
+	public String getReleaseDifferenceSummary() {
```

Comment:
Maybe `ReleaseDifferenceSummaryGenerator` should generate all this and not just modify what it returns.

Reply:
I'll pass in as arguments the existingRelease and replacing release's name/version to encapsulate this better.  Passing in the whole release object seems like a bit overkill...